### ISSUE TITLE
Is there an example of multiple checkboxes binded to a multi selection?

### ISSUE THREAD

**Issue by Kashdog** [altair-viz/altair#1115]
I know for a select binding for single selection you would do something like 

binding = alt.binding_select(options=['Graph','Original Graph', 'Trend'])
click = alt.selection_single(encodings=['color'], bind=binding)

How would a multi selection binding to a set of checkboxes look?

Thanks

**Comment by jakevdp** [comment]
You can create two selections, each with their own binding, and add both selections to the chart.

We don't have any example of that at the moment, but would be happy to accept a contribution!

**Comment by f4lco** [comment]
I believe I was looking for a similar use-case and would like to share the results of my research.

Originally I wanted to create a line chart, with the possibility to select a subset of lines. I thought of a multiple selection, either using an HTML input of type `<select multiple>`, or a set of checkboxes, would map to my problem.

To my belief, the approach to use a set of checkboxes with multi selection is not feasible due to the way checkboxes are supposed to work. Because checkboxes only emit boolean values, one would have to map each checkbox to a field value, or more generally: a set of boolean values to set of "allowed" field values for the resulting filter. I think this is out of the scope of the current `alt.selection_multi` intent / implementation.

As already mentioned, a _set of checkboxes_ is essentially equivalent to the `<select multiple>` approach in terms of affordance. However, true multi select appears to be work-in-progress: see vega/vega#1488 or vega/vega-lite#3722.

To sum up, using multi select with set of checkboxes appears to be difficult, and requires writing framework-level code instead of justing wiring existing components up correctly.

My current workaround is to draw a "clickable legend" in separate chart using `mark_rect`, whose input consists of dataframe just listing the possible field values. I use a condition of on  the color to highlight selected field values. Then I place a multi selection on the "legend" chart and reuse the selection in the actual line plot to paint only selected curves.
I consider everything needed to implement the described approach well documented. However, if an example is needed to spread the idea of how to emulate multi-select (sets of checkboxes)  in current Altair, I could provide one.

**Comment by TristanK27** [comment]
> I believe I was looking for a similar use-case and would like to share the results of my research.
> 
> Originally I wanted to create a line chart, with the possibility to select a subset of lines. I thought of a multiple selection, either using an HTML input of type `<select multiple>`, or a set of checkboxes, would map to my problem.
> 
> To my belief, the approach to use a set of checkboxes with multi selection is not feasible due to the way checkboxes are supposed to work. Because checkboxes only emit boolean values, one would have to map each checkbox to a field value, or more generally: a set of boolean values to set of "allowed" field values for the resulting filter. I think this is out of the scope of the current `alt.selection_multi` intent / implementation.
> 
> As already mentioned, a _set of checkboxes_ is essentially equivalent to the `<select multiple>` approach in terms of affordance. However, true multi select appears to be work-in-progress: see [vega/vega#1488](https://github.com/vega/vega/issues/1488) or [vega/vega-lite#3722](https://github.com/vega/vega-lite/issues/3722).
> 
> To sum up, using multi select with set of checkboxes appears to be difficult, and requires writing framework-level code instead of justing wiring existing components up correctly.
> 
> My current workaround is to draw a "clickable legend" in separate chart using `mark_rect`, whose input consists of dataframe just listing the possible field values. I use a condition of on the color to highlight selected field values. Then I place a multi selection on the "legend" chart and reuse the selection in the actual line plot to paint only selected curves.
> I consider everything needed to implement the described approach well documented. However, if an example is needed to spread the idea of how to emulate multi-select (sets of checkboxes) in current Altair, I could provide one.

Dear f4lco,

I know this is an old question. But could it still be possible to provide an example?

Kind regards
TristanK27

**Comment by f4lco** [comment]
Dear @TristanK27,

sure, here you go:

```python
import altair as alt
alt.renderers.enable('notebook')

import pandas as pd

make = pd.DataFrame({'name': ['Honda', 'Ford', 'Dodge']})
fuel = pd.DataFrame({
    'Honda': [9, 8, 8, 7, 7],
    'Ford': [5, 4, 3, 2, 1],
    'Dodge': [6, 5, 5, 3, 4]
}).reset_index().melt(id_vars=['index'], var_name='name', value_name='fuel')

selection = alt.selection_multi(fields=['name'])
color = alt.condition(selection, alt.Color('name:N'), alt.value('lightgray'))
make_selector = alt.Chart(make).mark_rect().encode(y='name', color=color).add_selection(selection)
fuel_chart = alt.Chart(fuel).mark_line().encode(x='index', y=alt.Y('fuel', scale=alt.Scale(domain=[0, 10])), color='name').transform_filter(selection)

make_selector | fuel_chart
```
![ezgif-6-8a9ab86e4b81](https://user-images.githubusercontent.com/29141485/78459388-e2d6a800-76b8-11ea-9cf3-8c4567174917.gif)

Selecting a single line works with a single click, and adding or removing a line works with shift-click. I hope it addresses / solves your problem. Enjoy!
